feat: remove deprecations

griest024 opened the pull request:
## PR Checklist

- [ ] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [ ] Tests added/updated (for bug fixes/features)
- [ ] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [ ] Bug fix
- [ ] Feature
- [ ] Style update
- [ ] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
<!-- Describe the current behavior and link to relevant issue -->

Fixes: # <!-- Closes issue on merge -->
Part of: # <!-- Keeps issue open -->

## New behavior
<!-- Describe the changes -->

## Breaking change?

- [ ] Yes
- [ ] No

<!-- If yes, describe impact and migration path -->

## Additional context
<!-- Any other relevant information -->

griest024 commented:
@xelaint can you review the design removals?

xelaint commented on apps/design-land/src/app/card/card.component.html:
```diff
@@ -2,7 +2,7 @@ <h1>Card</h1>
 
 <h2>Overview</h2>
-<p>There are three types of cards: default (filled), raised, and stroked. Cards can contain anything from images, blocks of text, lists, accordions, and other components.</p>
+<p>There are three types of cards: default (filled), and stroked. Cards can contain anything from images, blocks of text, lists, accordions, and other components.</p>
```
```suggestion
<p>There are two types of cards: default (filled) and stroked. Cards can contain anything from images, blocks of text, lists, accordions, and other components.</p>
```

xelaint commented on apps/design-land/src/app/sidebar/sidebar.component.html:
```diff
@@ -54,7 +54,7 @@ <h4>Required imports</h4>
 
 <h3>Header and footer</h3>
-<p>The <code>&lt;daff-sidebar-header&gt;</code> includes optional title (<code>[daffSidebarHeaderTitle]</code>) and action (<code>[daffSidebarHeaderAction]</code>) selectors, and a slot to render custom content. The action selector should be used along with the <code>&lt;daff-icon-button&gt;</code> (view <a routerLink="/button">Button Documentation</a>) to make sure that the action is positioned  ...
+<p>The <code>&lt;daff-sidebar-header&gt;</code> includes optional title (<code>[daffSidebarHeaderTitle]</code>) selector, and a slot to render custom content. The action selector should be used along with the <code>&lt;daff-icon-button&gt;</code> (view <a routerLink="/button">Button Documentation</a>) to make sure that the action is positioned correctly and it passes WCAG guidelines.</p>
```
```suggestion
<p>The <code>&lt;daff-sidebar-header&gt;</code> includes an optional title (<code>[daffSidebarHeaderTitle]</code>) selector and a slot to render custom content. The action selector should be used along with the <code>&lt;daff-icon-button&gt;</code> (view <a routerLink="/button">Button Documentation</a>) to make sure that the action is positioned correctly and it passes WCAG guidelines.</p>
```

xelaint commented on libs/design-examples/card/src/linkable-card/linkable-card.component.html:
```diff
@@ -18,20 +18,6 @@ <h4 daffCardTitle>Basel Exhibition Centre</h4>
-  </div>
-</a>
-
 <select [formControl]="raisedColorControl">
```
this select can be removed.

griest024 committed:
feat(design)!: remove `DaffNotificationSubtitleDirective` (#4271)

BREAKING CHANGE: `DaffNotificationSubtitleDirective` has been removed. Use the `DaffNotificationMessageDirective` instead.

griest024 committed:
feat(design)!: remove `DaffRaisedCardComponent` (#4271)

BREAKING CHANGE: `DaffRaisedCardComponent` has been removed. Use the `elevated` property of `daff-card` instead.

griest024 committed:
feat(design)!: remove `DaffSidebarHeaderActionDirective` (#4271)

BREAKING CHANGE: `DaffSidebarHeaderActionDirective` has been removed. Use the `dismissible` property of `DaffSidebarHeaderComponent` instead.

griest024 committed:
feat(driver)!: remove `DaffDriverMagentoModule` (#4271)

BREAKING CHANGE: `DaffDriverMagentoModule` has been removed. Use `provideMagentoDriver` instead.
